### ISSUE TITLE
Upload mixed paste content

### ISSUE THREAD

**Issue by brhoades** [brhoades/pasty-http-client#5]
Nontrivial rewrite required.

It should be possible to upload a paste that contains binary files and code files.  Currently you must choose between one or the other. Obviously images and binary files came first; it's the simplest and only supports one file. This particular schema inhibits pasty-cli from handling a list of files; instead it has to upload a "code" paste or a "binary" paste.

Current objects:
*Code files*
```javascript
{
  files: [
    {
      id: 1,                          // order in a list
      name: "filename",
      contents: "raw text contents",
      type: "hljs file type"  // "auto" or "js", "cpp", etc
     },
     {
       id: 2,
       ...
     }
   ],
   type: "code"    
}
```

*Images / binary file*
```javascript
{
  name: "filename",
  data: "base64 encoded data",
  mime: "file mime",
  type: "file"
}
```

Requirements:
 * Parsers and serializers should be moved to pasty-core
 * Versioned
 * Mixed content support
 * Differentiate base64-encoded binary files from raw text.

**Comment by brhoades** [comment]
This issue is more of a pasty-core issue. It inhibits pasty-cli from being easier to use. Rather than simply giving pasty-cli a list of files to upload and letting it do what needs to be done, a type for the paste must be selected.

**Comment by brhoades** [comment]
```
{
  files: [
    {
      id: 1,
      name: "filename",
      data: "code code code",
      meta: {
        mime: "text/plain",
        highlight: "auto"
       }
     },
     {
       id: 2,
       name: "image.png",
       data: "base64encodeddata",
       meta: {
         mime: "image/png"
        }
     },
     {
       id: 3,
       name: "",
       data: "base64encodeddata",
       meta: {
         mime: "application/octet-stream"
        }
     }
   ],
   version: 2
}
```

**Comment by brhoades** [comment]
Implemented as part of pasty-core 0.1.0 and pasty-http-client v0.3.0
